File: src/CattyTests/NextLookBrickTests.m
//
//  NextLookBrickTests.m
//  Catty
//
//  Created by Marc Slavec on 5/22/14.
//
//

#import <XCTest/XCTest.h>
#import "BrickTests.h"

@interface NextLookBrickTests : BrickTests

@end

@implementation NextLookBrickTests

- (void)setUp
{
    [super setUp];
    // Put setup code here. This method is called before the invocation of each test method in the class.
}

- (void)tearDown
{
    // Put teardown code here. This method is called after the invocation of each test method in the class.
    [super tearDown];
}

-(void)testNextLookBrick
{
    SpriteObject* object = [[SpriteObject alloc] init];
    Program *program = [Program defaultProgramWithName:@"a"];
    object.program = program;
    
    NSBundle *bundle = [NSBundle bundleForClass:[self class]];
    NSString * filePath = [bundle pathForResource:@"test.png"
                                           ofType:nil];
    NSData *imageData = UIImagePNGRepresentation([UIImage imageWithContentsOfFile:filePath]);
    Look* look = [[Look alloc] initWithName:@"test" andPath:@"test.png"];
    [imageData writeToFile:[NSString stringWithFormat:@"%@images/%@", [object projectPath], @"test.png"]atomically:YES];
    Look* look1 = [[Look alloc] initWithName:@"test2" andPath:@"test2.png"];
    [imageData writeToFile:[NSString stringWithFormat:@"%@images/%@", [object projectPath], @"test2.png"]atomically:YES];
    
    NextLookBrick* brick = [[NextLookBrick alloc] init];
    brick.object = object;
    [object.lookList addObject:look];
    [object.lookList addObject:look1];
    object.currentLook = look;
    object.currentUIImageLook = [UIImage imageWithContentsOfFile:filePath];
    object.currentLookBrightness = 0.0f;

    
    
    dispatch_block_t action = [brick actionBlock];
    
    action();
    XCTAssertEqual(object.currentLook,look1, @"NextLookBrick not correct");
    [Program removeProgramFromDiskWithProgramName:program.header.programName];
}
@end

File: src/CattyTests/Tests/Bricks/NextLookBrickTests.swift
/**
 *  Copyright (C) 2010-2019 The Catrobat Team
 *  (http://developer.catrobat.org/credits)
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Affero General Public License as
 *  published by the Free Software Foundation, either version 3 of the
 *  License, or (at your option) any later version.
 *
 *  An additional term exception under section 7 of the GNU Affero
 *  General Public License, version 3, is available at
 *  (http://developer.catrobat.org/license_additional_term)
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 *  GNU Affero General Public License for more details.
 *
 *  You should have received a copy of the GNU Affero General Public License
 *  along with this program.  If not, see http://www.gnu.org/licenses/.
 */

import XCTest

@testable import Pocket_Code

final class NextLookBrickTests: AbstractBrickTests {

    func testNextLookBrick() {
        let object = SpriteObject()
        let project = Project.defaultProject(withName: "a", projectID: "")
        let spriteNode = CBSpriteNode(spriteObject: object)
        object.spriteNode = spriteNode
        object.project = project

        let bundle = Bundle(for: NextLookBrickTests.self)
        let filePath = bundle.path(forResource: "test.png", ofType: nil)
        let imageData: Data? = UIImage(contentsOfFile: filePath ?? "")!.pngData()
        let look = Look(name: "test", andPath: "test.png")
        try? imageData?.write(to: URL(fileURLWithPath: "\(object.projectPath()!)images/\("test.png")"), options: [.atomic])
        let look1 = Look(name: "test2", andPath: "test2.png")
        try? imageData?.write(to: URL(fileURLWithPath: "\(object.projectPath()!)images/\("test2.png")"), options: [.atomic])

        let script = WhenScript()
        script.object = object

        let brick = NextLookBrick()
        brick.script = script
        object.lookList.add(look as Any)
        object.lookList.add(look1 as Any)
        spriteNode.currentLook = look
        spriteNode.currentUIImageLook = UIImage(contentsOfFile: filePath ?? "")

        let action: () -> Void = brick.actionBlock()

        action()
        XCTAssertEqual(spriteNode.currentLook, look1, "NextLookBrick not correct")
        Project.removeProjectFromDisk(withProjectName: project.header.programName, projectID: project.header.programID)
    }
}
